### ISSUE TITLE
Hyphenation for other languages

### ISSUE THREAD

**Issue by tobyShandy** [mbutterick/hyphenate#3]
After looking at the code (rather superficially I admit), am I right to assume that converting the TeX hyphenation patterns into hashed-patterns would allow me to hypenate other languages using Pollen?

And if so, would it be possible then to have a settable option at the pollen.rkt level to load alternate hashed-pattern lists?

**Comment by mbutterick** [comment]
No, a set of hyphenation patterns is language-specific, and the patterns in `hyphenate` are only good for English.

However, pattern files for other languages are available (e.g., in the [CTAN archive](https://www.ctan.org/tex-archive/language)) which could be plugged into `hyphenate` in the manner you suggest. 

The main hurdle is verifying that the pattern files can be redistributed as part of the `hyphenate` package under the LGPL. That is true for the original Knuth–Plass patterns but I haven’t investigated others.

If so, yes, it would be easy to make that a settable option.

**Comment by tobyShandy** [comment]
OK. The situation is more or less as I imagined, though I hadn't given any serious thought to the legal aspect until you mentioned it.

So, I've done some investigation and it seems all right:
I wanted to hyphenate French material and located the patterns on the [TUG website](https://www.tug.org/tex-hyphen/) (much more synoptic view of the available resources than CTAN). The file containing the patterns is free to use and only stipulates that its name has to be changed in case of modification.
I've also located a more recent, extended list of patterns on [Dicollecte.org (in French)](http://www.dicollecte.org/download.php?prj=fr). It is devised for Open Office and released under the LGPL. 

I went ahead with the extended list and obtained [the following result (gist)](https://gist.github.com/tobyShandy/87574ce34d7389906560) through a bit of ugly racket scripting.

From there, I'm waiting for your input to know how to move any further.

PS: just noticed that I forgot greetings in my opening comment. Sorry about that, I didn't mean to sound blunt.

**Comment by tobyShandy** [comment]
OK. The situation is more or less as I imagined, though I hadn't given any serious thought to the legal aspect until you mentioned it.

So, I've done some investigation and it seems all right:
I wanted to hyphenate French material and located the patterns on the [TUG website](https://www.tug.org/tex-hyphen/) (much more synoptic view of the available resources than CTAN). The file containing the patterns is free to use and only stipulates that its name has to be changed in case of modification.
I've also located a more recent, extended list of patterns on [Dicollecte.org (in French)](http://www.dicollecte.org/download.php?prj=fr). It is devised for Open Office and released under the LGPL. 

I went ahead with the extended list and obtained [the following result (gist)](https://gist.github.com/tobyShandy/87574ce34d7389906560) through a bit of ugly racket scripting.

From there, I'm waiting for your input to know how to move any further.

PS: just noticed that I forgot greetings in my opening comment. Sorry about that, I didn't mean to sound blunt.

**Comment by tobyShandy** [comment]
Damn, sorry for the closing/ re-opening snafu. I'm really not used to github.

**Comment by mbutterick** [comment]
> I've also located a more recent, extended list of patterns on Dicollecte.org (in French). It is devised for Open Office and released under the LGPL.

That sounds fine.

I’m happy to add this capability. I just need to figure out how to add languages to the interface in the least surprising way.

**Comment by mbutterick** [comment]
This is almost ready.

**Comment by mbutterick** [comment]
See https://github.com/mbutterick/hyphenate/tree/dev-multilang

**Comment by mbutterick** [comment]
Closed by 8cb7f7413a4c442152594e4ac2c36a102d865ffa.

You can now use `hyphenate/fr` to access the French hyphenation library.

**Comment by mbutterick** [comment]
As in:

```racket
(require hyphenate/fr)
(hyphenate "C’est formidable, mes amis")
```
